otx/core/data/anomaly_dataset_adapter.py

```diff
@@ -0,0 +1,220 @@
+            id=ID(1),
+            name="Anomalous",
+            domain=self.domain,
+            is_anomalous=True,
+        )
+        return [normal_label, abnormal_label]
+
+    def convert_to_otx_format(self, datumaro_dataset: dict) -> Tuple[DatasetEntity, LabelSchemaEntity]:
+        """Convert DatumaroDataset to DatasetEntity for Anomalytasks."""
+        raise NotImplementedError
+
+
+class AnomalyClassificationDatasetAdapter(AnomalyBaseDatasetAdapter):
+    """Anomaly classification adapter inherited from AnomalyBaseDatasetAdapter."""
+
+    def convert_to_otx_format(self, datumaro_dataset: dict) -> Tuple[DatasetEntity, LabelSchemaEntity]:
+        """Convert DatumaroDataset to DatasetEntity for Anomaly classification."""
+        normal_label, abnormal_label = self._prepare_anomaly_label_information()
+        label_schema = self._generate_default_label_schema([normal_label, abnormal_label])
+
+        # Prepare
+        dataset_items = []
+        for subset, subset_data in datumaro_dataset.items():
+            for _, datumaro_items in subset_data.subsets().items():
+                for datumaro_item in datumaro_items:
+                    image = Image(file_path=datumaro_item.media.path)
+
+                    label = normal_label if datumaro_item.id.split("/")[0] == "good" else abnormal_label
+                    shapes = [
+                        Annotation(
+                            Rectangle.generate_full_box(),
+                            labels=[ScoredLabel(label=label, probability=1.0)],
+                        )
+                    ]
+                    # Unlabeled dataset
+                    annotation_scene = None  # type: Any
+                    if len(shapes) == 0:
+                        annotation_scene = NullAnnotationSceneEntity()
+                    else:
+                        annotation_scene = AnnotationSceneEntity(
+                            kind=AnnotationSceneKind.ANNOTATION, annotations=shapes
+                        )
+                    dataset_item = DatasetItemEntity(image, annotation_scene, subset=subset)
+                    dataset_items.append(dataset_item)
+
+        return DatasetEntity(items=dataset_items), label_schema
+
+
+class AnomalyDetectionDatasetAdapter(AnomalyBaseDatasetAdapter):
+    """Anomaly detection adapter inherited from AnomalyBaseDatasetAdapter."""
+
+    def convert_to_otx_format(self, datumaro_dataset: dict) -> Tuple[DatasetEntity, LabelSchemaEntity]:
+        """Conver DatumaroDataset to DatasetEntity for Anomaly detection."""
+        normal_label, abnormal_label = self._prepare_anomaly_label_information()
+        label_schema = self._generate_default_label_schema([normal_label, abnormal_label])
+
+        # Prepare
+        dataset_items = []
+        for subset, subset_data in datumaro_dataset.items():
+            for _, datumaro_items in subset_data.subsets().items():
+                for datumaro_item in datumaro_items:
```

Comment:
It would be nice if this part (for statement) was also subtracted as a function.